File: food_coon_app/lib/models/order_item.dart
class OrderItem {
  final int? id;
  final int orderId;
  final int menuItemId;
  final int quantity;
  final double unitPrice;
  final String notes;

  OrderItem({
    this.id,
    required this.orderId,
    required this.menuItemId,
    required this.quantity,
    required this.unitPrice,
    this.notes = '',
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      id: json['id'] as int?,
      orderId: json['orderId'] as int,
      menuItemId: json['menuItemId'] as int,
      quantity: json['quantity'] as int,
      unitPrice: (json['unitPrice'] as num).toDouble(),
      notes: json['notes'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
        if (id != null) 'id': id,
        'orderId': orderId,
        'menuItemId': menuItemId,
        'quantity': quantity,
        'unitPrice': unitPrice,
        'notes': notes,
      };
}

File: food_coon_rn/src/models/OrderItem.ts
export interface OrderItem {
  id?: number;
  orderId: number;
  menuItemId: number;
  quantity: number;
  unitPrice: number;
  notes: string;
}

export function orderItemToJson(item: OrderItem): Record<string, any> {
  return {
    orderId: item.orderId,
    menuItemId: item.menuItemId,
    quantity: item.quantity,
    unitPrice: item.unitPrice,
    notes: item.notes,
  };
}
